MaxText/convert_llama_ckpt.py

```diff
@@ -0,0 +1,346 @@
+
+
+  self_attention['query']['kernel'] = np.array(self_attention['query']['kernel'])
+  self_attention['key']['kernel'] = np.array(self_attention['key']['kernel'])
+  self_attention['value']['kernel'] = np.array(self_attention['value']['kernel'])
+  self_attention['out']['kernel'] = np.array(self_attention['out']['kernel'])
+  self_attention['query']['kernel'] = np.transpose(self_attention['query']['kernel'],axes=(1, 0, 2, 3))
+  self_attention['key']['kernel'] = np.transpose(self_attention['key']['kernel'],axes=(1, 0, 2, 3))
+  self_attention['value']['kernel'] = np.transpose(self_attention['value']['kernel'],axes=(1, 0, 2, 3))
+  #layers, base_num_query_heads * head_dim, base_num_query_heads, head_dim =>
+  #base_num_query_heads, layers,head_dim, base_num_query_heads * head_dim
+  self_attention['out']['kernel'] = np.transpose(self_attention['out']['kernel'],axes=(2, 0, 3, 1))
+
+  #scale the query weights
+  self_attention['query']['kernel'] = self_attention['query']['kernel']/np.sqrt(head_dim)
+
+  jax_weights['decoder']['decoder']['self_attention'] = self_attention
+
+
+
+
+
+  layer_weight['mlp']['wi_0']['kernel'] = np.array(layer_weight['mlp']['wi_0']['kernel'])
+  layer_weight['mlp']['wi_1']['kernel'] = np.array(layer_weight['mlp']['wi_1']['kernel'])
+  layer_weight['mlp']['wo']['kernel'] = np.array(layer_weight['mlp']['wo']['kernel'])
+  layer_weight['pre_self_attention_layer_norm']['scale'] = np.array(layer_weight['pre_self_attention_layer_norm']['scale'])
+  layer_weight['post_self_attention_layer_norm']['scale'] = np.array(layer_weight['post_self_attention_layer_norm']['scale'])
+  #swap the layer index
+  layer_weight['mlp']['wi_0']['kernel'] = np.transpose(layer_weight['mlp']['wi_0']['kernel'],axes=(1, 0, 2))
+  layer_weight['mlp']['wi_1']['kernel'] = np.transpose(layer_weight['mlp']['wi_1']['kernel'],axes=(1, 0, 2))
+  layer_weight['mlp']['wo']['kernel'] = np.transpose(layer_weight['mlp']['wo']['kernel'],axes=(1, 0, 2))
+  layer_weight['pre_self_attention_layer_norm']['scale'] = np.transpose(
+                                    layer_weight['pre_self_attention_layer_norm']['scale'],
+                                    axes=(1, 0))
+  layer_weight['post_self_attention_layer_norm']['scale'] = np.transpose(
+                                    layer_weight['post_self_attention_layer_norm']['scale'],
+                                    axes=(1, 0))
+
+  jax_weights['decoder']['decoder']['mlp'] = layer_weight['mlp']
+  jax_weights['decoder']['decoder']['pre_self_attention_layer_norm'] = layer_weight['pre_self_attention_layer_norm']
+  jax_weights['decoder']['decoder']['post_self_attention_layer_norm'] = layer_weight['post_self_attention_layer_norm']
+
+  #convert all weights to jax.numpy
+  jax_weights = jax.tree_map(jnp.array, jax_weights)
+
+  print(f"jax_weights = {jax_weights}")
+
+  #dummy configs for the checkpoint_manager
+  step_number_to_save_new_ckpt = 0
```

Comment:
Why are we involving MaxText at all at this point? It seems brittle. Can we just create an orbax checkpoint manager and checkpoint?

Reply:
removed - thank you!